src/main/java/com/moneylendingapp/annotations/CurrentUser.java

```diff
@@ -0,0 +1,11 @@
+package com.moneylendingapp.annotations;
+
+import org.springframework.security.core.annotation.AuthenticationPrincipal;
+
+import java.lang.annotation.*;
+
+@Retention(RetentionPolicy.RUNTIME)
```

Comment:
Is this really different from using @Principal or @AuthenticationPrincipal?
I am not seeing the different thing this interface is doing